### ISSUE TITLE
Failed to analyze archives: Could not find the corresponding model element for use

### ISSUE THREAD

**Issue by vmassol** [revapi/revapi#39]
Hi,

On the XWiki project we're having a build failure when building https://github.com/xwiki/xwiki-platform/tree/955505595b95784876777a7059fb3bc725c01776/xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-oldcore

We're getting:

`````
[INFO] --- revapi-maven-plugin:0.4.5:check (revapi-check) @ xwiki-platform-legacy-oldcore ---
Downloading: http://nexus.xwiki.org/nexus/content/groups/public/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.pom
Downloading: https://repo.maven.apache.org/maven2/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.pom
Downloading: http://forge.octo.com/archiva/repository/3rd-party/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.pom
Downloading: http://repository.atlassian.com/maven2/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.pom
Downloading: http://svn.atlassian.com/svn/public/contrib/repository/maven2/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.pom
Downloading: http://nexus.xwiki.org/nexus/content/groups/public/javax/ejb/ejb/2.0/ejb-2.0.jar
Downloading: http://nexus.xwiki.org/nexus/content/groups/public/javax/faces/jsf-api/1.1/jsf-api-1.1.jar
Downloading: http://nexus.xwiki.org/nexus/content/groups/public/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.jar
Downloading: https://repo.maven.apache.org/maven2/javax/ejb/ejb/2.0/ejb-2.0.jar
Downloading: https://repo.maven.apache.org/maven2/javax/faces/jsf-api/1.1/jsf-api-1.1.jar
Downloading: https://repo.maven.apache.org/maven2/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.jar
Downloading: http://forge.octo.com/archiva/repository/3rd-party/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.jar
Downloading: http://repository.atlassian.com/maven2/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.jar
Downloading: http://svn.atlassian.com/svn/public/contrib/repository/maven2/com/sun/speech/freetts/freetts/1.2.1/freetts-1.2.1.jar
[WARNING] Failed to resolve dependencies of old artifacts: Resolution of some artifacts failed: Could not locate the file of the maven artifact: javax.faces:jsf-api:jar:1.1, Could not locate the file of the maven artifact: com.sun.speech.freetts:freetts:jar:1.2.1, Failed to read artifact descriptor for com.sun.speech.freetts:freetts:jar:1.2.1, Could not locate the file of the maven artifact: javax.ejb:ejb:jar:2.0.. The API analysis might produce unexpected results.
[WARNING] Failed to resolve dependencies of new artifacts: Resolution of some artifacts failed: Could not locate the file of the maven artifact: javax.faces:jsf-api:jar:1.1, Could not locate the file of the maven artifact: com.sun.speech.freetts:freetts:jar:1.2.1, Could not locate the file of the maven artifact: javax.ejb:ejb:jar:2.0.. The API analysis might produce unexpected results.
[INFO] Comparing [org.xwiki.platform:xwiki-platform-legacy-oldcore:jar:8.2.1] against [org.xwiki.platform:xwiki-platform-legacy-oldcore:jar:8.3-SNAPSHOT] (including their transitive dependencies).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:12 min
[INFO] Finished at: 2016-08-19T15:55:09+02:00
[INFO] Final Memory: 122M/984M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.revapi:revapi-maven-plugin:0.4.5:check (revapi-check) on project xwiki-platform-legacy-oldcore: Failed to analyze archives: Could not find the corresponding model element for use: RawUseSite[siteClass='com.xpn.xwiki.internal.MessageToolVelocityContextInitializer', siteDescriptor='null', siteName='null', sitePosition=-1, siteType=CLASS, useType=IS_IMPLEMENTED] of type org.xwiki.velocity.VelocityContextInitializer -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.revapi:revapi-maven-plugin:0.4.5:check (revapi-check) on project xwiki-platform-legacy-oldcore: Failed to analyze archives
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to analyze archives
	at org.revapi.maven.Analyzer.analyze(Analyzer.java:309)
	at org.revapi.maven.AbstractRevapiMojo.analyze(AbstractRevapiMojo.java:199)
	at org.revapi.maven.CheckMojo.execute(CheckMojo.java:42)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: java.lang.IllegalStateException: Could not find the corresponding model element for use: RawUseSite[siteClass='com.xpn.xwiki.internal.MessageToolVelocityContextInitializer', siteDescriptor='null', siteName='null', sitePosition=-1, siteType=CLASS, useType=IS_IMPLEMENTED] of type org.xwiki.velocity.VelocityContextInitializer
	at org.revapi.java.compilation.ProbingEnvironment.toUseSite(ProbingEnvironment.java:349)
	at org.revapi.java.compilation.ProbingEnvironment.access$000(ProbingEnvironment.java:68)
	at org.revapi.java.compilation.ProbingEnvironment$1.visit(ProbingEnvironment.java:168)
	at org.revapi.java.compilation.ProbingEnvironment.visitRawUseSites(ProbingEnvironment.java:212)
	at org.revapi.java.compilation.ProbingEnvironment.visitUseSites(ProbingEnvironment.java:160)
	at org.revapi.java.JavaElementDifferenceAnalyzer.appendUses(JavaElementDifferenceAnalyzer.java:312)
	at org.revapi.java.JavaElementDifferenceAnalyzer.endAnalysis(JavaElementDifferenceAnalyzer.java:249)
	at org.revapi.Revapi.analyze(Revapi.java:238)
	at org.revapi.Revapi.analyzeWith(Revapi.java:188)
	at org.revapi.Revapi.analyze(Revapi.java:119)
	at org.revapi.maven.Analyzer.analyze(Analyzer.java:307)
	... 24 more
[ERROR] 
`````

Any idea?

Thanks
-Vincent

**Comment by vmassol** [comment]
Hi Lukas

Just to let you kow, this is a critical issue for us since it forces us to disable revapi (and thus backward compat checking) since there doesn't see much that we can do at our level.

Do you think you could look at it relatively quickly?

Anything I can do to help?

Thanks!

**Comment by tmortagne** [comment]
Some more details:
* com.xpn.xwiki.internal.MessageToolVelocityContextInitializer is supposed to be exceluded (like any class in an "internal" package)
* this class was deleted from xwiki-platform-oldcore module which is bundled in xwiki-platform-legacy-oldcore module using aspectj-maven-plugin

**Comment by metlos** [comment]
I'll let you know when I find something. It seems like Revapi is not finding the classes it should...

The exclusion of a class happens only after the classpath usage model is built so that's why we are probably seeing this even for an excluded class. 

Your second point, @tmortagne, might be on to something, but I don't know yet.

The meaning of the error message is that Revapi found `org.xwiki.velocity.VelocityContextInitializer` to implement `com.xpn.xwiki.internal.MessageToolVelocityContextInitializer` but it could not find the latter class in its classpath model.

**Comment by metlos** [comment]
With commit  5a12221 I am able to successfully check the oldcore of xwiki but I need to investigate the test failures that I get in revapi's testsuite.

But now I need to crash to bed :wink:

**Comment by vmassol** [comment]
Thanks so much for your help Lukas and sorry to keep you awake so late! We owe you a new beer (it's going to be a whole cask now!) for the time when we'll meet IRL in the future :)

**Comment by metlos** [comment]
Hopefully, this is now fixed. If you find a problem with it, please open a new issue. Thanks for reporting it!

**Comment by vmassol** [comment]
Thank you Lukas. Yes it seems to be working now! :)
